app/main/routes.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+from app import mongo
+from flask import Blueprint, current_app, render_template
+
+main = Blueprint("main", __name__)
+
+# database collection variables
+desserts_collection = mongo.db.desserts
+recipes_collection = mongo.db.recipes
+
+#----- Global Helper -----#
+def get_total_recipes():
+        return int(recipes_collection.count())
+@main.context_processor
+def total_recipes():
+        return dict(total_recipes=get_total_recipes)
+
+
+#----- HOME -----#
+@main.route("/")
+def home():
+        carousel = recipes_collection.aggregate([{"$sample": {"size": 8}}])
+        
+        # get recipes for random recipe link
+        random_recipe = recipes_collection.aggregate([{"$sample": {"size": 1}}])
+
+        # get desserts for random dessert link
+        categories = []
+        for dessert in desserts_collection.find().sort([("desserts", 1)]):
+                dessert_name = dessert.get("dessert_type")
+                for item in dessert_name:
+                        categories.append(item)

```

Comment:
There should be a more efficient way to do this. Since you need only one category in the end, just get a random one from the `categories_collection` or use the category that `random_recipe` belongs to. You don't need to get the random collection from the templates.
 Avoid double loops.

Reply:
Same here! Genius! One line of code now in Python, with minor tweaks in Jinja to get a |random item from the carousel instead! Thanks for that!